src/reader/decompress.rs

```diff
@@ -63,14 +63,27 @@ pub enum CompressionType {
     Zstd,
 }
 
-fn decode_header(bytes: &[u8]) -> (bool, usize) {
-    let a: [u8; 3] = (&bytes[..3]).try_into().unwrap();
-    let a = [0, a[0], a[1], a[2]];
-    let length = u32::from_le_bytes(a);
-    let is_original = a[1] & 1 == 1;
-    let length = (length >> (8 + 1)) as usize;
+/// Indicates length of block and whether it's compressed or not.
+#[derive(Debug, PartialEq, Eq)]
+enum CompressionHeader {
+    Original(u32),
+    Compressed(u32),
+}
 
-    (is_original, length)
+/// ORC files are compressed in blocks, with a 3 byte header at the start
+/// of these blocks indicating the length of the block and whether it's
+/// compressed or not.
+fn decode_header(bytes: [u8; 3]) -> CompressionHeader {
+    let bytes = [bytes[0], bytes[1], bytes[2], 0];
+    let length = u32::from_le_bytes(bytes);
+    let is_original = length & 1 == 1;
+    // to clear the is_original bit
+    let length = length >> 1;
+    if is_original {
+        CompressionHeader::Original(length)
+    } else {
+        CompressionHeader::Compressed(length)
+    }
```

Comment:
Just trying to make things more explicit, utilize Rust's enums more instead of relying on booleans